internal/services/confluent/confluent_organization_resource.go

```diff
@@ -0,0 +1,500 @@
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/confluent/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ConfluentOrganizationResource struct{}
+
+type ConfluentOrganizationResourceModel struct {
+	Name              string                      `tfschema:"name"`
+	ResourceGroupName string                      `tfschema:"resource_group_name"`
+	Location          string                      `tfschema:"location"`
+	OfferDetail       []ConfluentOfferDetailModel `tfschema:"offer_detail"`
+	UserDetail        []ConfluentUserDetailModel  `tfschema:"user_detail"`
+	LinkOrganization  []ConfluentLinkOrgModel     `tfschema:"link_organization"`
+	Tags              map[string]string           `tfschema:"tags"`
+
+	OrganizationId    string `tfschema:"organization_id"`
+	SsoUrl            string `tfschema:"sso_url"`
+	CreatedTime       string `tfschema:"created_time"`
+	ProvisioningState string `tfschema:"provisioning_state"`
+}
+
+type ConfluentOfferDetailModel struct {
+	Id              string   `tfschema:"id"`
+	PlanId          string   `tfschema:"plan_id"`
+	PlanName        string   `tfschema:"plan_name"`
+	PublisherId     string   `tfschema:"publisher_id"`
+	TermUnit        string   `tfschema:"term_unit"`
+	PrivateOfferId  string   `tfschema:"private_offer_id"`
+	PrivateOfferIds []string `tfschema:"private_offer_ids"`
+	TermId          string   `tfschema:"term_id"`
+	Status          string   `tfschema:"status"`
+}
+
+type ConfluentUserDetailModel struct {
+	EmailAddress      string `tfschema:"email_address"`
+	FirstName         string `tfschema:"first_name"`
+	LastName          string `tfschema:"last_name"`
+	AadEmail          string `tfschema:"aad_email"`
+	UserPrincipalName string `tfschema:"user_principal_name"`
+}
+
+type ConfluentLinkOrgModel struct {
+	Token string `tfschema:"token"`
+}
+
+func (r ConfluentOrganizationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"offer_detail": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"plan_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
```

Comment:
Please add validation for possible values for `plan_id`. If SDK / OpenAPI spec does not provide this, please add a `func PossibleValuesForPlanId() []string` in the services/confluent/validate package